ext/fileinfo/fileinfo.c

```diff
@@ -376,8 +376,8 @@ static void _php_finfo_get_type(INTERNAL_FUNCTION_PARAMETERS, int mode, int mime
 				break;
 
 			default:
-				php_error_docref(NULL, E_WARNING, "Can only process string or stream arguments");
-				RETURN_FALSE;
+				zend_argument_type_error(2, "must be of type string|resource, %s given", zend_zval_type_name(what));
```

Comment:
I think this message is good as-is, although probably the order of the types could be swapped. As far as I saw, `object`, `array` are both displayed before strings, so maybe it's worth to do the same with resources too.